pyproject.toml

```diff
@@ -33,9 +33,10 @@ dependencies = [
   "graphviz>=0.18.2, <0.21",
   "halo>=0.0.31, <0.1.0",
   "more-itertools>=8.10.0, <10.2.0",
-  "pandas>=1.5.0, <1.6.0",
+  # pandas 2.2 has a bug https://github.com/databricks/databricks-sql-python/issues/334, so using 2.1.
+  "pandas>=2.1, <2.2",
```

Comment:
This is super restrictive, I don't think we can block 1.x installs.

Medium term I'd like to just remove pandas as a production dependency altogether, we don't need it for anything.